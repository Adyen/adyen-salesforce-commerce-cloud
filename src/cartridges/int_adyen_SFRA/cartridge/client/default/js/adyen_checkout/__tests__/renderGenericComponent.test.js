/**
 * @jest-environment jsdom
 */
const { renderGenericComponent } = require('../renderGenericComponent');
const { createSession } = require('../../commons');
const { fetchGiftCards } = require('../../commons');
const store = require('../../../../../store');

jest.mock('../../commons');
jest.mock('../../../../../store');

beforeEach(() => {
  window.AdyenCheckout = jest.fn(async () => ({
    create: jest.fn(),
    paymentMethodsResponse: {
      storedPaymentMethods: [{ supportedShopperInteractions: ['Ecommerce'] }],
      paymentMethods: [{ type: 'amazonpay' }],
    },
    options: {
      amount: 'mocked_amount',
      countryCode: 'mocked_countrycode',
    },
  }));
  window.Configuration = { amount: 0 };
  store.checkoutConfiguration = {

  };
  store.checkout = {
   options: {}
  };
  createSession.mockReturnValue({
    adyenConnectedTerminals: { uniqueTerminalIds: ['mocked_id'] },
    id: 'mock_id',
    sessionData: 'mock_session_data',
    imagePath: 'example.com',
    adyenDescriptions: {},
  });

  fetchGiftCards.mockReturnValue({
    giftCards: [
      {
        orderAmount: {
          currency: '',
          value: '',
        },
      },
    ],
  });
});
describe('Render Generic Component', () => {
  it('should render', async () => {
    document.body.innerHTML = `
      <div id="paymentMethodsList"></div>
      <input type="radio" name="brandCode" value="card" />
      <button value="submit-payment">Submit</button>
      <div id="component_card"></div>
      <div id="adyenPosTerminals">
        <span>Child #1</span>
      </div>
      <div>
        <input type="text" id="shippingFirstNamedefault" value="test">
        <input type="text" id="shippingLastNamedefault" value="test">
        <input type="text" id="shippingAddressOnedefault" value="test">
        <input type="text" id="shippingAddressCitydefault" value="test">
        <input type="text" id="shippingZipCodedefault" value="test">
        <input type="text" id="shippingCountrydefault" value="test">
        <input type="text" id="shippingPhoneNumberdefault" value="test">
        <input type="text" id="shippingZipCodedefault" value="test">
      </div>
    `;

    store.componentsObj = { foo: 'bar', bar: 'baz' };
    store.checkoutConfiguration.paymentMethodsConfiguration = { amazonpay: {} };
    await renderGenericComponent();
    expect(createSession).toBeCalled();
    expect(store.checkoutConfiguration).toMatchSnapshot();
    expect(
        document.querySelector('input[type=radio][name=brandCode]').value,
    ).toBeTruthy();
  });
});
