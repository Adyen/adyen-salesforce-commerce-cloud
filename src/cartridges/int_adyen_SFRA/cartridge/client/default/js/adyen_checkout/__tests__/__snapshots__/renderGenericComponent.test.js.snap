// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Generic Component should render 1`] = `
{
  "amount": "mocked_amount",
  "countryCode": "mocked_countrycode",
  "paymentMethodsConfiguration": {
    "amazonpay": {
      "addressDetails": {
        "addressLine1": "test",
        "city": "test",
        "countryCode": "test",
        "name": "test test",
        "phoneNumber": "test",
        "postalCode": "test",
        "stateOrRegion": "test",
      },
      "configuration": undefined,
    },
  },
  "session": {
    "adyenDescriptions": {},
    "id": "mock_id",
    "imagePath": "example.com",
    "sessionData": "mock_session_data",
  },
}
`;
